internal/broadcaster/broadcaster.go

```diff
@@ -39,62 +39,60 @@ func Run(ctx context.Context, cfg *config.Config) {
 	log.Info("Starting service")
 
 	r := &Runner{
-		log:          log,
-		participants: data.NewParticipantsQ(cfg.DB().Clone()),
-		Broadcaster:  cfg.Broadcaster(),
+		log:         log,
+		q:           data.NewAirdropsQ(cfg.DB().Clone()),
+		Broadcaster: cfg.Broadcaster(),
 	}
 
 	running.WithBackOff(ctx, r.log, "builtin-broadcaster", r.run, 5*time.Second, 5*time.Second, 5*time.Second)
 }
 
 func (r *Runner) run(ctx context.Context) error {
-	participants, err := r.participants.New().FilterByStatus(data.TxStatusPending).Limit(r.QueryLimit).Select()
+	airdrops, err := r.q.New().FilterByStatus(data.TxStatusPending).Limit(r.QueryLimit).Select()
 	if err != nil {
-		return fmt.Errorf("select participants: %w", err)
+		return fmt.Errorf("select airdrops: %w", err)
 	}
-	if len(participants) == 0 {
+	if len(airdrops) == 0 {
 		return nil
 	}
-	r.log.Debugf("Got %d participants to broadcast airdrop transactions", len(participants))
+	r.log.Debugf("Got %d pending airdrops, broadcasting now", len(airdrops))
 
-	for _, participant := range participants {
-		log := r.log.WithField("participant_nullifier", participant.Nullifier)
-
-		if err := r.handleParticipant(ctx, participant); err != nil {
-			log.WithError(err).Error("Failed to handle participant")
+	for _, drop := range airdrops {
+		if err = r.handlePending(ctx, drop); err != nil {
+			r.log.WithField("airdrop", drop).
+				WithError(err).Error("Failed to handle pending airdrop")
 			continue
 		}
 	}
 
 	return nil
 }
 
-func (r *Runner) handleParticipant(ctx context.Context, participant data.Participant) error {
-	tx, err := r.createAirdropTx(ctx, participant)
-	if err != nil {
-		return fmt.Errorf("creating airdrop tx: %w", err)
-	}
+func (r *Runner) handlePending(ctx context.Context, airdrop data.Airdrop) (err error) {
+	var txHash string
 
-	txHash, err := r.broadcastTx(ctx, tx)
-	if err != nil {
-		err = r.participants.New().UpdateStatus(participant.Nullifier, txHash, data.TxStatusFailed)
+	defer func() {
 		if err != nil {
-			return fmt.Errorf("update participant failed tx status: %w", err)
```

Comment:
log error

Reply:
it is logged in `handlePending`, this would be double log